WordPress/gutenberg
# Styles panel: Rearrange the Revisions and Additional CSS toggles

Fixes https://github.com/WordPress/gutenberg/issues/66307
Fixes https://github.com/WordPress/gutenberg/issues/66333

## What?

The way to access Styles revisions and Additional CSS is inconsistent and little discoverable, especially for the Additional CSS. These features live in sub-panels of the Styles 'drill-down' menu. Other sub-panels use standard navigation buttons with a chevron icon to access the sub-panels.

For more details, please refer to the conversations on https://github.com/WordPress/gutenberg/issues/66307
and https://github.com/WordPress/gutenberg/issues/66333.

## Why?

For consistency and better user experiences, accessing the sub-panels of the Styles drill-down menu should use a consistent pattern. Features should be well discoverable and not buried down into an ellipsis menu.

## How?

- Removes the Revisions icon button from the panel header.
- Adds a 'Styles revisions' navigation button at the bottom of the panel. This button should appear only when there are revisions.
- Removes the 'Additional CSS' menu item from the ellipsis 'More' menu.
- Makes the 'Additional CSS' navigation button in the panel always visible, also when there's no custom CSS set. This basically reverts https://github.com/WordPress/gutenberg/issues/47266
- Removes the descriptions for 'Blocks' and 'Additional CSS'. 

Additionally: Fixes a bug where accessing the Additional CSS from the navigation button at the bottom of the Styles panel
did not switch back the canvas to the default view when the Style Book was enabled. See https://github.com/WordPress/gutenberg/issues/66333#issuecomment-2435162060

## Testing Instructions




- Make sure there is no additional CSS set.
- Make sure there are no Styles revisions. I'm not sure how to do this other than deleting all the posts of type `revision` and title `Custom Styles` from the database.
- Go to the Site editor > Styles panel in the edit view.
- Observe there is an 'Additional CSS' button at the bottom of the Styles panel.
- Observe there's no 'Styles revisions' button shown, yet.
- Make any style change and save.
- Navigate back to the root of the panel and observe there is now a 'Styles revisions' button shown.
- Test navigating to the 'Styles revisions' panel with and without 'Style book' enabled and verify the canvas view changes as expected. There should be no differences in the behavior compared to trunk
- Enable the Style Book, navigate to Additional CSS and verify that it does deactivate the Style Book.
- Click 'Open Navigation' at the top left of the screen.
- Observe that in the Design > Styles panel there is no 'revisions' navigation item after 'Blocks' and 'Additional CSS'. It's not necessary to show it because there is alreadh a 'Revisions' navigation button in the root navigation panel (the black one).

Edit:
- Go to the Site editor > Styles panel in the view mode, that is: the new Styles panel on the left.
- Observe there's no 'Styles revisions' button.

### Testing Instructions for Keyboard


## Screenshots or screencast 

Before and after:

![before and after](https://github.com/user-attachments/assets/31e220b7-9cb9-4dbf-a552-26b1fd231f12)
